src/pybind/mgr/cephadm/module.py

```diff
@@ -2525,6 +2548,11 @@ def tuned_profile_rm_setting(self, profile_name: str, setting: str) -> str:
         self._kick_serve_loop()
         return f'Removed setting {setting} from tuned profile {profile_name}'
 
+    @handle_orch_error
+    def service_discovery_dump_cert(self) -> str:
+        root_cert = self.get_store('service_discovery/root/cert')
+        return root_cert if root_cert else 'No certificate found for service discovery'
```

Comment:
Given this ends up showing in the terminal (via `HandleCommandResult()`), what about using stdout for printing the cert, stderr for the error message, and returning an non-zero error code?

Reply:
fixed by raising an Exception, stdout is the return value